### PR TITLE
sdp: fix FF PT change

### PR DESCRIPTION
**Pull request by algesten**

### CODE REVIEW

**Comment by k0nserv** [comment]
Following on my comments in #281, this code only runs when we add a new m-line, but we also need to handle this on direction changes I think? For example, what about an m-line that's initially offered as `inactive` and is later made `recvonly` or `sendonly`

File: src/change/sdp.rs [hunk 1]
```diff
@@ -800,7 +800,9 @@ fn add_new_lines(
             media.need_open_event = need_open_event;
 
             // Match/narrow remote params.
-            session.codec_config.update_params(&m.rtp_params());
+            session
+                .codec_config
+                .update_params(&m.rtp_params(), m.direction());
```
Comment (k0nserv) [inline src/change/sdp.rs:805]: The direction needs to be inverted here since `Media::direction` returns the local direction. Same in `update_media`

**Review by k0nserv — COMMENTED** [review]
The logic around `sendonly` and the Pts being suggestions for what the remote wants to send only apply for offers. Do we have a bug here when we have offered a `recvonly` m-line and are processing the answer(in which case remote_dir will be `sendonly`)

File: src/change/sdp.rs (hunk 1 above)
Comment (algesten) [reply src/change/sdp.rs:805]: This is m (remote m-line). Not media.

File: src/format.rs [hunk 2]
```diff
@@ -587,8 +613,15 @@ impl CodecConfig {
         }
 
         // Now lock potential new parameters to remote.
+        //
+        // If the remote is doing `SendOnly`, the PTs are suggestions, and we are allowed to
+        // ANSWER with our own allocations as overrides. If SendRecv or RecvOnly, the remote
+        // is talking about its own receiving capapbilities and we are not allowed to change it
+        // in the ANSWER.
+        let warn_on_locked = !remote_dir.is_receiving();
```
[the remark this answers is not in the document]
Comment (algesten) [reply src/format.rs]: Inactive _with_ codecs? 🤔

File: src/change/sdp.rs (hunk 1 above)
Comment (k0nserv) [reply src/change/sdp.rs:805]: Ah, so it is

**Review by k0nserv — APPROVED** [review]

**Pull request merged**